<h1><center>DATA ANALYSIS REPORT</center></h1>

##### BY: Asimiea Nicholas Walson

After cleaning and merging the three dataset into a master dataset, it was saved to a csv file named **"twitter_archive_master.csv"**.

The following steps were taken to draw insight from the data.

**Read the master dataset into a pandas dataframe named df_master**

In [14]:
# Import libraries required for project
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import os
%matplotlib inline

# Indicate the name of folder where project data is located
folder_name = 'data'

In [15]:
# Read master dataset into pandas dataframe
df_master = pd.read_csv(os.path.join(folder_name, 'twitter_archive_master.csv'))

In [16]:
# Show numeric information about dataset
df_master.describe()

tweet_id  rating_numerator  rating_denominator      img_num  \
count  1.686000e+03       1686.000000         1686.000000  1686.000000   
mean   7.392710e+17         11.473903           10.534401     1.215896   
std    6.800607e+16          8.106793            6.945891     0.576191   
min    6.660209e+17          0.000000            2.000000     1.000000   
25%    6.773835e+17         10.000000           10.000000     1.000000   
50%    7.132943e+17         11.000000           10.000000     1.000000   
75%    7.931770e+17         12.000000           10.000000     1.000000   
max    8.921774e+17        165.000000          150.000000     4.000000   

           p1_conf      p2_conf       p3_conf  retweet_count  favorite_count  
count  1686.000000  1686.000000  1.686000e+03    1686.000000     1686.000000  
mean      0.605150     0.137147  6.135847e-02    2817.118031     9233.109727  
std       0.266416     0.101252  5.188051e-02    4807.199770    12583.246963  
min       0.044333     0.000010  2.160900e-07      16.000000       81.000000  
25%       0.378746     0.054938  1.598680e-02     649.250000     2163.250000  
50%       0.606701     0.121432  4.995100e-02    1437.500000     4440.000000  
75%       0.851578     0.198429  9.468462e-02    3247.750000    11656.750000  
max       0.999984     0.467678  2.734190e-01   79515.000000   132810.000000

Establish the source of most of the tweets in the dataframe

In [7]:
df_master.source.value_counts()

<a href="http://twitter.com/download/iphone" rel="nofollow">Twitter for iPhone</a>     1655
<a href="http://twitter.com" rel="nofollow">Twitter Web Client</a>                       22
<a href="https://about.twitter.com/products/tweetdeck" rel="nofollow">TweetDeck</a>       9
Name: source, dtype: int64

Establish the percentage of the highest tweet source

In [8]:
df_master.source.value_counts()[0] / len(df_master.index) * 100

98.16132858837486

Confirm the highest occuring rating numerator

In [9]:
df_master.rating_numerator.mode()

0    12
dtype: int64

Confirm the highest occuring rating denominator

In [10]:
df_master.rating_denominator.mode()

0    10
dtype: int64

Establish the most popular dog breed from p1, p2 and p3 columns respectively

In [11]:
df_master.p1.value_counts()

golden_retriever      139
labrador_retriever     95
pembroke               88
chihuahua              79
pug                    54
                     ... 
toyshop                 1
hay                     1
cougar                  1
syringe                 1
carton                  1
Name: p1, Length: 214, dtype: int64

In [12]:
df_master.p2.value_counts()

labrador_retriever          96
golden_retriever            82
cardigan                    72
chihuahua                   43
chesapeake_bay_retriever    40
                            ..
swing                        1
japanese_spaniel             1
cradle                       1
bow_tie                      1
basketball                   1
Name: p2, Length: 229, dtype: int64

In [13]:
df_master.p3.value_counts()

labrador_retriever    76
chihuahua             57
golden_retriever      46
eskimo_dog            36
kelpie                33
                      ..
television             1
bagel                  1
buckeye                1
tripod                 1
tiger_cat              1
Name: p3, Length: 234, dtype: int64

### Insights:
1. 98% of the tweets were made using an iPhone 

2. Most frequent rating numerator is 12 and most frequent rating demonimator is 10.

3. Most popular dog breeds are golden retriever followed by the labrador retriever before the chihuahua.

### Visualization

**Question**: Is there a relationship between retween count and favourite count?

![scatter](scatter.png)

The plot above attempts to visualize the relationship between retweet_count and favorite_count.
We can see there is a positive linear relationship between the two series which implies the tendency of increment in either retweet_count as favorite_count increases and vice versa. This would not always be, but most likely be the case.